web/src/app/users/linkToMobile/VerifyCodeFields.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from 'react'
+import {
+  DialogContent,
+  DialogContentText,
+  Grid,
+  TextField,
+  makeStyles,
+} from '@material-ui/core'
+import gql from 'graphql-tag'
+import { useMutation } from 'react-apollo'
+
+const useStyles = makeStyles({
+  codeContainer: {
+    marginTop: '2em',
+  },
+  contentText: {
+    marginBottom: 0,
+  },
+  textField: {
+    textAlign: 'center',
+    fontSize: '1.5rem',
+  },
+})
+
+const mutation = gql`
+  mutation($input: VerifyAuthLinkInput!) {
+    verifyAuthLink(input: $input)
+  }
+`
+
+interface VerifyCodeFieldsProps {
+  authLinkID: string
+}
+
+export default function VerifyCodeFields(
+  props: VerifyCodeFieldsProps,
+): JSX.Element {
+  const classes = useStyles()
+  const [numOne, setNumOne] = useState('')
+  const [numTwo, setNumTwo] = useState('')
+  const [numThree, setNumThree] = useState('')
+  const [numFour, setNumFour] = useState('')
+
+  const [verifyCode] = useMutation(mutation, {
+    variables: {
+      input: {
+        id: props.authLinkID,
+        code: numOne + numTwo + numThree + numFour,
+      },
+    },
+  })
+
+  useEffect(() => {
+    if (numOne && numTwo && numThree && numFour) {
+      verifyCode()
+    }
+  }, [numOne, numTwo, numThree, numFour])
+
+  return (
+    <DialogContent>
+      <Grid container spacing={2}>
+        <Grid item xs={12}>
+          <DialogContentText className={classes.contentText}>
+            Enter the code displayed on your mobile device.
+          </DialogContentText>
+        </Grid>
+        <Grid
+          className={classes.codeContainer}
+          item
+          xs={12}
+          container
+          spacing={2}
+        >
+          <Grid item xs={3}>
+            <TextField
+              id='numOne'
+              value={numOne}
+              onChange={(e) => {
+                setNumOne(e.target.value)
+                if (!numTwo) {
+                  const inputTwo = document.getElementById('numTwo')
+                  if (inputTwo) inputTwo.focus()
+                }
+              }}
+              inputProps={{
+                maxLength: 1,
+                className: classes.textField,
+              }}
+            />
+          </Grid>
+          <Grid item xs={3}>
+            <TextField
+              id='numTwo'
+              value={numTwo}
+              onChange={(e) => {
+                setNumTwo(e.target.value)
+                if (!numThree) {
+                  const inputThree = document.getElementById('numThree')
+                  if (inputThree) inputThree.focus()
+                }
+              }}
+              inputProps={{
+                maxLength: 1,
+                className: classes.textField,
+              }}
+            />
+          </Grid>
```

Comment:
Since these are all identical, we can just render them in a loop for whatever length we want/need entry to be. For the focus bit just use the index in the id (e.g. `verify-${idx}`)

The conditional on the "next" element will already handle the final field since `document.getElementById(verify-${idx+1})` will return null.